src/assets/wise5/classroomMonitor/classroomMonitorComponents/shared/grading-helpers/grading-helpers.ts

```diff
@@ -1,10 +1,24 @@
-export function isComponentVisibleToStudent(nodeStatus: any, component: any): boolean {
-  let result = true;
-  if (nodeStatus.componentStatuses != null) {
-    const componentStatus = nodeStatus.componentStatuses[component.id];
-    if (componentStatus != null) {
-      result = componentStatus.isVisible;
-    }
+import { ComponentStatus } from '../../../../common/ComponentStatus';
+import { TeacherProjectService } from '../../../../services/teacherProjectService';
+
+export function calculateComponentVisibility(
+  nodeId: string,
+  componentStatuses: { [componentId: string]: ComponentStatus },
+  projectService: TeacherProjectService
+): { [componentId: string]: boolean } {
+  const componentIdToIsVisible = {};
+  for (const component of projectService.getComponents(nodeId)) {
+    componentIdToIsVisible[component.id] =
+      projectService.componentHasWork(component) &&
+      isComponentVisibleToStudent(componentStatuses, component.id);
   }
-  return result;
+  return componentIdToIsVisible;
+}
```

Comment:
I'm not sure that passing in the entire TeacherProjectService to this function is a good idea, since doing this exposes all of TeacherProjectService's public functions to this function. This function just needs the components and componentsWithWork for the node. What if we re-wrote it to something like this? I think it'll make it easier to test too (remember to write tests)

```
export function calculateComponentVisibility(
  componentStatuses: { [componentId: string]: ComponentStatus },
  components: ComponentContent[],
  componentsWithWork: ComponentContent[]
): { [componentId: string]: boolean } {
```